src/main/java/org/folio/qm/service/impl/FieldProtectionSetterServiceImpl.java

```diff
@@ -93,14 +93,17 @@ private boolean isAnyIndicator2InSettingOrIndicator2Match(MarcFieldProtectionSet
 
   private boolean isAnySubFieldInSettingOrSubFieldMatch(MarcFieldProtectionSetting setting, FieldItem field) {
     return setting.getSubfield().equals(ANY_STRING)
-      || !Pattern.compile("[$]" + setting.getSubfield()).matcher(field.getContent().toString()).matches();
+      || Pattern.compile("[$]" + setting.getSubfield() + "\\b").matcher(field.getContent().toString()).find();
```

Comment:
Will it work if a setting has protected $b and field content is: $atest1$btest2 ?

Reply:
no it won't. Is it a valid scenario to set subfield values without whitespaces?

Reply:
@oliinyko yes, it's possible:
![image](https://user-images.githubusercontent.com/5904625/183850604-ea33268a-d071-4bc9-95af-4233c8e08056.png)

Reply:
ok, I've removed word boundary from pattern